anchor/anchorRecord.go

```diff
@@ -17,22 +17,29 @@ import (
 	"github.com/FactomProject/factomd/common/primitives"
 )
 
-//AnchorRecord is used to construct anchor chain
+// AnchorRecord is used to construct the anchor chain. The Factom Protocol writes an anchor into
+// a parent blockchain (Bitcoin or Ethereum) approximately every 10 minutes. The accumulated entries
+// from the previous 10 minutes are organized into a Directory Block (DBlock). The information
+// from the directory block is stored in the AnchorRecord.
 type AnchorRecord struct {
-	AnchorRecordVer int
-	DBHeight        uint32 `json:",omitempty"` // The only directory block height included in this anchor
-	KeyMR           string `json:",omitempty"` // Merkle root of the only directory block included in this anchor
+	AnchorRecordVer int    // version 1 places signature content in the field, version 2 uses external IDs for signature
+	DBHeight        uint32 // Factom Directory Block Height - the unique number associated with this DBlock
+	KeyMR           string // key merkle root of the directory block
```

Comment:
Does the removal of `json:",omitempty"` change the behavior here? If yes, why is safe and what's the benefit of the change? If not, why apply this removal to only those 2 fields and not the rest?

Reply:
Ahh! Good catch, this was an error on merging. Sam had made some changes and I screwed that up. Fixed

Reply:
Note, i also fixed the DBHeight as well, which i had also merged poorly.